server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -53,9 +55,73 @@
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.common.settings.Setting.intSetting;
 
 public class SniffConnectionStrategy extends RemoteConnectionStrategy {
 
+    /**
+     * A list of initial seed nodes to discover eligible nodes from the remote cluster
+     */
+    public static final Setting.AffixSetting<List<String>> REMOTE_CLUSTER_SEEDS_OLD = Setting.affixKeySetting(
+            "cluster.remote.",
+            "seeds",
+            key -> Setting.listSetting(
+                    key,
+                    Collections.emptyList(),
+                    s -> {
+                        // validate seed address
+                        parsePort(s);
+                        return s;
+                    },
+                    Setting.Property.Dynamic,
+                    Setting.Property.NodeScope));
+
+    /**
+     * A list of initial seed nodes to discover eligible nodes from the remote cluster
+     */
+    public static final Setting.AffixSetting<List<String>> REMOTE_CLUSTER_SEEDS = Setting.affixKeySetting(
+        "cluster.remote.",
+        "sniff.seeds",
+        key -> Setting.listSetting(key,
```

Comment:
There is a listSetting overload that allows providing a fallback setting. Why not use that?

Reply:
I don't follow your comment here. The listSetting there is taking a fallback setting? I followed the example of what we did for the `search.remote` -> `cluster.remote` transition here.

Reply:
It's the lines below this comment that confuse me, in particular that regexp magic. I wonder why that is necessary.

Reply:
The regex is necessary because "key" is not the namespace. It is the settings key. So in the non-"_na_" case this will be "cluster.remote.cluster_name.sniff.seeds". The regex produces the proper fallback key "cluster.remote.cluster_name.seeds". The example for this is in the `ScopedSettingsTests#testDependentSettingsWithFallback` test.

Reply:
ok, thanks for the explanation.